mantidproject/mantid
# Improve performance of colorfill plot of large workspaces

**Description of work.**
This PR improves the performance of colorfill plots for large workspaces by reducing the amount of data that is initially calculated for large workspaces and resampling the data when needed.

**To test:**
Load a large workspace e.g. `WISH00040503` from the system test data
Do a colorfill plot
The plot should appear in less than 10 seconds.
Check it looks correct
Check all the functionality still works correctly

Fixes #28693 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
